UACloudLibraryServer/PostgreSQLDB.cs

```diff
@@ -304,10 +304,17 @@ private string[] FindNodesetsInTable(string[] keywords, string tableName)
                     }
 
                     NpgsqlCommand sqlCommand = new NpgsqlCommand(sqlInsert, _connection);
-                    object result = sqlCommand.ExecuteScalar();
-                    if ((result != null) && !results.Contains(result.ToString()))
+                    NpgsqlDataReader reader = sqlCommand.ExecuteReader();

```

Comment:
The sql Reader should be closed after completing the function. You could put a using(...) around the Command and the Reader.

Reply:
Fixed.